ouroboros-network/src/Ouroboros/Network/PeerSelection/Simple.hs

```diff
@@ -52,34 +54,49 @@ withPeerSelectionActions localRootTracer publicRootTracer timeout readTargets
     let peerSelectionActions = PeerSelectionActions {
             readPeerSelectionTargets = readTargets,
             readLocalRootPeers = toList <$> readTVar localRootsVar,
-            requestPublicRootPeers = requestLedgerPeers,
+            requestPublicRootPeers = requestLedgerPeers dnsActions,
             requestPeerGossip = \_ -> pure [],
             peerStateActions
           }
+        dnsActions = DNSActions {
+            dnsResolverResource = resolverResource,
+            dnsAsyncResolverResource = asyncResolverResource,
+#if defined(mingw32_HOST_OS)
+            dnsNewResolverResource = newResolverResource,
+#endif
+            dnsLookupAWithTTL = lookupAWithTTL
+          }
```

Comment:
The same here, if we had access to `ioDNSActions` they could be plugged here too.